tcms/rpc/api/forms/testrun.py

```diff
@@ -68,7 +68,18 @@ def clean_status(self):
         return data
 
 
-class UpdateExecutionForm(BaseCaseRunForm):
-    tested_by = forms.ModelChoiceField(queryset=User.objects.all(), required=False)
-    build = forms.ModelChoiceField(queryset=Build.objects.all(), required=False)
-    case_text_version = forms.CharField(required=False)
+def validate_case_text_version(value):
+    value = value.strip()
+    if not value.isdigit() and not value == 'latest':
+        raise forms.ValidationError('Invalid case_text_version. Number or latest supported.')
```

Comment:
See how IntegerField does its validation (method to_python) + https://docs.djangoproject.com/en/3.0/ref/forms/validation/. 

It is a bit different from what you do here, mostly allowing for localization from what I can figure out.   Also because the update() method below will be changing I'm not sure if we can go with a CharField or should instead look into a custom IntegerField where -1 will mean latest.